### PR TITLE
Fix check for vehicle in active range

### PR DESCRIPTION
**Pull request by Stephen2**
#### Summary

SUMMARY: None

#### Purpose of change

The code here was was
```
...actually just wrong. You can't iterate over a two-dimensional array like this. It's iterating over whole one-dimensional arrays, which decay to pointer type and are then cast to bool. So every bool will be true, which is almost certainly not intended.
```
See comment history for the evolution of this PR

#### Describe the solution

Looks like this commit (the last commit in blame on this code before this one): a84902e
introduced the change to array access, but really wasn't intending to change the functionality

I brought back the original functionality

#### Describe alternatives you've considered

None

#### Testing

Compiles and launches

### CODE REVIEW

**Comment by EvgenijM86** [comment]
You have removed reference from the loop, so now it hard copies it instead? If yes, I don't think that this is a better approach. You probably won't notice a difference on bool variable, but on larger objects it will be noticeable.

**Comment by olanti-p** [comment]
>Are there any blockers from using clang10 in the CI process?

Travis is set up to run a clang10 build, but due to failing tests / time limit it pretty much never gets a chance
(e.g. right now for this PR https://travis-ci.org/github/CleverRaven/Cataclysm-DDA/builds/704599261 )

>You have removed reference from the loop, so now it hard copies it instead? If yes, I don't think that this is a better approach.

I disagree. Copying a fundamental value and using the copy is cheaper that making a pointer to that fundamental value and using it via a pointer.

**Comment by Stephen2** [comment]
I'll be honest guys, I truly don't now C++ and barely understand pointers, pointer access costs, memory copy costs etc...  I'll need someone smarter than me to determine if this code change is OK or not.

This is why my job title says "Manager" these days :laughing:

**Comment by jbytheway** [comment]
I think the code is actually just wrong.  You can't iterate over a two-dimensional array like this.  It's iterating over whole one-dimensional arrays, which decay to pointer type and are then cast to bool.  So every bool will be `true`, which is almost certainly not intended.

**Comment by Stephen2** [comment]
@jbytheway :oof: looks like you're 100% right hah!

I'll try and decipher what the code was trying to do, and get back to you

**Comment by Stephen2** [comment]
Looks like this commit (the last commit in blame on this code before this one):
https://github.com/CleverRaven/Cataclysm-DDA/commit/a84902e1bef3f247dbbacaf933ecb9e97b4c4fe0
introduced the change to array access, but really wasn't intending to change the functionality

I brought back the original functionality

**Comment by Stephen2** [comment]
OK, ready for review @jbytheway /anyone

File: src/map.cpp [hunk 1]
```diff
@@ -450,17 +451,18 @@ bool map::vehproceed( VehicleList &vehicle_list )
...
-                if( has_vehicle ) {
-                    still_active = true;
-                    break;
-                }
-            }
-        }
-        cache.veh_in_active_range = still_active;
+
+        // Check if any vehicles exist in the active range for this z-level
+        // Short-circuit the evaluation if no "veh_in_active_range" for performance
```
Comment (jbytheway) [inline src/map.cpp:456]:
```suggestion
```
Don't think this comment is necessary.

**Pull request merged**